motor-control-module/src/main/java/org/sagebionetworks/research/motor_control_module/show_step_fragment/hand_selection/ShowHandSelectionStepFragment.java

```diff
@@ -0,0 +1,149 @@
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.sagebionetworks.research.motor_control_module.show_step_fragment.hand_selection;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.graphics.drawable.Drawable;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v7.widget.DividerItemDecoration;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import org.sagebionetworks.research.domain.result.AnswerResultType;
+import org.sagebionetworks.research.domain.result.ResultType;
+import org.sagebionetworks.research.domain.result.implementations.AnswerResultBase;
+import org.sagebionetworks.research.domain.result.interfaces.AnswerResult;
+import org.sagebionetworks.research.mobile_ui.perform_task.PerformTaskFragment;
+import org.sagebionetworks.research.mobile_ui.show_step.view.ShowStepFragmentBase;
+import org.sagebionetworks.research.mobile_ui.show_step.view.ShowFormUIStepFragment;
+import org.sagebionetworks.research.motor_control_module.R;
+import org.sagebionetworks.research.presentation.model.form.ChoiceInputFieldViewBase;
+import org.sagebionetworks.research.presentation.model.form.InputFieldView;
+import org.sagebionetworks.research.presentation.model.interfaces.StepView;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.threeten.bp.Instant;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ShowHandSelectionStepFragment extends ShowFormUIStepFragment {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ShowHandSelectionStepFragment.class);
+    public static final String HAND_SELECTION_KEY = "handSelection";
+    public static final String HAND_ORDER_KEY = "handOrder";
+
+    @NonNull
+    public static ShowHandSelectionStepFragment newInstance(@NonNull StepView stepView) {
+        ShowHandSelectionStepFragment fragment = new ShowHandSelectionStepFragment();
+        Bundle arguments = ShowStepFragmentBase.createArguments(stepView);
+        fragment.setArguments(arguments);
+        return fragment;
+    }
+
+    public void writeHandSelectionResult(@HandSelection String handSelection) {
```

Comment:
This logic should go in the View Model. It should go in the Step's view model. This fragment currently doesn't talk to a Step view model but talks to its parent's view model